crates/crypto/proof-params/src/lib.rs

```diff
@@ -31,22 +73,32 @@ pub mod spend {
 }
 
 /// Proving key for the output proof.
-#[cfg(feature = "proving-keys")]
-pub static OUTPUT_PROOF_PROVING_KEY: Lazy<ProvingKey<Bls12_377>> =
-    Lazy::new(proving_keys::output_proving_parameters);
+pub static OUTPUT_PROOF_PROVING_KEY: Lazy<LazyProvingKey> = Lazy::new(|| {
+    let mut output_proving_key = LazyProvingKey::new();
 
-/// Proving key for the spend proof.
+    #[cfg(all(feature = "proving-keys", not(target_arch = "wasm32")))]
```

Comment:
Here and elsewhere I don't think we want to have any arch-specific `cfg`s.

Rather, we should either:

- have the `proving-keys` feature enabled, in which case the proving key is bundled into the binary (regardless of architecture)
- have the `proving-keys` feature not enabled, in which case either:
  - a byte slice with the proving key data is passed to an initialization function before proving, or
  - nothing is passed before proving, causing a panic.
  
  This behavior shouldn't depend on architecture, though, it should just be code configuration.

Reply:
Also, maybe we should rename the feature to `bundled-proving-keys`?

Reply:
The `proving-keys` feature in the wasm crate is intrinsically enabled since its complex chain of dependencies require it. I'm trying to think through an optimal design approach here that caters to both configurations. 

Can we somehow make the behavior depend on the source crate compiling the code rather than the target architecture? For instance, if we're building the binary from the wasm crate, it'll detect that and conditionally skip the code block above. Something like `#[cfg(all(feature = "proving-keys", not(env = "wasm_crate")))]`. 

Marking the `proving-keys` feature not enabled with `#[cfg(not(feature = "proving-keys"))]` would require changing the existing calls (https://github.com/penumbra-zone/penumbra/blob/main/crates/bin/pcli/tests/proof.rs#L33) to something like `let pk = &SPEND_PROOF_PROVING_KEY.load(byte_slice)`. Ideally we shouldn't need to make code modifications outside the `proof-params` package.

Reply:
Right now, various library crates need to enable the `proving-keys` feature, because it currently controls whether those items _exist at all_ in the source code. With the change I suggested, (let's assume we did the rename), we'd have a feature `bundled-proving-keys` that controls whether the proving keys are _pre-initialized with bundled data_.

In that world, there is no need for any library to toggle the feature on, and nothing that requires the wasm crate to have it on.  Rather, libraries would never enable the feature, and _binaries_ (the roots of the dependency graph) would add a dependency on the `proof-params` crate with the `bundled-proving-keys` feature enabled. Since features are unified across the entire dep tree, this allows individual binaries to opt in to bundled proving keys or not.